### PR TITLE
26214: Bundle Product / Create / Price validation for dynamic price #27409

### PR DESCRIPTION
**Pull request by phuocnh**
### Description (*)
- Fixed Bundle Product / Create / Price validation for dynamic price

### Fixed Issues (if relevant)

1. magento/magento2#26214: Bundle Product / Create / Price validation for dynamic price

### Manual testing scenarios (*)
1. Enter "New Product / Bundle" form
2. Turn off "Dynamic price"
3. Enter the price and remove it. You'll see a validation error.
4. Turn on "Dynamic price". You'll see remove a validation error.

video fixed https://youtu.be/zyPJw6udqO0

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by phuocnh** [comment]
@magento give me test instance

**Comment by magento-engcom-team** [comment]
Hi @phuocnh. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @phuocnh, here is your [new Magento instance](https://pr-27858.instances.magento-community.engineering).
Admin access: [https://pr-27858.instances.magento-community.engineering/admin_eebc](https://pr-27858.instances.magento-community.engineering/admin_eebc)
Login: `75e3b391` Password: `b88b92d5a72e`
Instance will be terminated in up to 3 hours.

**Comment by phuocnh** [comment]
@magento the new Magento instance working fine with code fixed

**Comment by phuocnh** [comment]
Hi @magento

I have problem when install MFTF test with branch **2.4-develop** and flow by document link https://devdocs.magento.com/mftf/docs/getting-started.html. I think it only support with Magento 2.3.x



thanks

**Comment by rogyar** [comment]
Hi @phuocnh. Have you added `.env` file as described on [step 2](https://devdocs.magento.com/mftf/docs/getting-started.html#environment-settings)?

**Comment by phuocnh** [comment]
hi @rogyar 

Yes, .env have content

{code}
MAGENTO_BASE_URL=http://magentoc24.test/
MAGENTO_BACKEND_NAME=admin
MAGENTO_ADMIN_USERNAME=phuoc
MAGENTO_ADMIN_PASSWORD=admin123
BROWSER=chrome
MODULE_WHITELIST=Magento_Framework,ConfigurableProductWishlist,ConfigurableProductCatalogSearch
ELASTICSEARCH_VERSION=7
{code}

I think have problem with magento 2.4-develop

thanks

**Comment by rogyar** [comment]
It might be something environment-related. I've recently configured the MFTF execution based on the latest `2.4-develop` branch with no issues. 

If you take a look at vendor/codeception/codeception/src/Codeception/Configuration.php:183, you will see that this error appears if the Codeception merged configuration equals the default configuration. So, literally, when the system is not configured for a specific environment for some reason.

I hope that this direction will give you more clues.

File: app/code/Magento/Ui/view/base/web/js/lib/core/element/links.js [hunk 1]
```diff
@@ -79,6 +79,10 @@ define([
 
         component.set(property, value, owner);
 
+        if (property === 'disabled' && value) {
```
Comment (gabrieldagama) [inline app/code/Magento/Ui/view/base/web/js/lib/core/element/links.js:82]: I believe `updateValue` shouldn't change the `validate` property when an element receives the 'disabled' property, that may introduce undesirable behaviour to other components. Instead, I would recommend a change more towards the component itself recognizes that is being disabled and update the validation.

As an example, `app/code/Magento/Ui/view/base/web/js/form/element/abstract.js` has the validation method that considers if the element is disabled, so basically we need to validate the component again if the component is disabled, that will undo the validation issue.

**Review by gabrieldagama — CHANGES_REQUESTED** [review]
Hi @phuocnh, thanks for the pull request! Please follow some comments on it, let me know if you have any questions.

**Comment by gabrieldagama** [comment]
Hi @phuocnh, can you have a look at my comment above? Please let me know if you can address the issues mentioned there.

Thanks.

**Comment by gabrieldagama** [comment]
Closing this PR due to inactivity, please feel free to reopen it. Thanks!

**Pull request closed without merging**